CONTRIBUTING.md

```diff
@@ -29,30 +31,138 @@ and maintained by Google](https://opensource.google.com/docs/cla/#why), not to
      being made to the model, whilst also ensuring that course-corrections can be
      made early in the process. If a test is growing to more than 500 lines, it
      may need to be broken into multiple smaller tests.
 
-1. The pull request should include both the suggested feature profile textproto 
+1. The pull request should include both the suggested feature profile textproto
  additions, as well as any relevant additions to tests. Tests should be written
  in golang using the [ONDATRA](https://github.com/openconfig/ondatra) framework.
 
 1. The automated CI running against each pull request will check the pull
  request for compliance.  The author should resolve any issues found by CI.
 
-1. One or more peers in the community may review the pull request.   
+1. One or more peers in the community may review the pull request.
 
 1. A feature profile repository maintainer will be reponsible for a final review
-and approval.  Only a feature repository maintainer can merge a pull request to 
+and approval.  Only a feature repository maintainer can merge a pull request to
 the main branch.
   
-The aim of this process is maintain the model quality and approach that OpenConfig 
+The aim of this process is maintain the model quality and approach that OpenConfig
 working group has strived for since its inception in 2014. Thank you for your contributions!
+
+## Test development style guide
+
+### Directory Organization
+
+The directory tree is organized as follows:
+
+* `cloudbuild/` contains google cloud build scripts for running virtual
+    routers in containers on [KNE](https://github.com/google/kne)
+* `feature/` contains definition and tests of feature profiles.
+* `internal/` contains packages used by feature profile tests.
+* `proto/`  contains protobuf files for feature profiles.
+* `tools/` contains code used for CI checks.
+* `topologies/` contains the testbed topology definitions.
+* `yang/` contains autogenerated go structures.
+
+### Test Suite Organization
+
+Under `feature/<featurename>/[<sub-feature>/]tests`, each feature group has its own
+subdirectory. For example:
+
+* `feature/bgp/` container for BGP feature profiles and test suites.
+* `feature/bgp/addpath` contains feature profiles  related to BGP.
+* `feature/bgp/addpath/tests` contains test code related to BGP Add Path
+    features.
+* `feature/lldp/` contains feature profiles for ISIS.
+* `feature/lldp/tests` contains test suite related to LLDP.
+* `internal/deviations` contains code which overrides standard tests where
+    there are known bugs.
+
+Under each feature group, the `*.go` file should be named after an appropriate
+[github featureprofiles project](https://github.com/orgs/openconfig/projects/2/views/1?filterQuery=)
+item. For example:
+
+* `tests/interfaces/rt_5_1_singleton_test.go` implements issue **RT-5.1
+  Singleton Interface**.
+* `tests/interfaces/rt_5_2_aggregate_test.go` implements issue **RT-5.2
```

Comment:
I don't think `interfaces` is specific enough for this example.  `tests/aggregate_interfaces/aggregate_interfaces_test.go` makes more sense to me.  The full path would be something like `feature/interfaces/tests/aggregate_interfaces/aggregate_interfaces_test.go`.

Reply:
Also just noticed comment from @sthesayi in another PR - the examples should be singular and not plural.

Reply:
Let me know what you think of the latest update.  Do you feel strongly about dropping the test suite identifier (ie: rt_5_1) from the test suite name?